src/rat/plugins/forecasting.py

```diff
@@ -341,6 +339,373 @@ def get_GFS_data(basedate, lead_time, basin_bounds, gfs_dir):
+            forecast_outflow[f'ST_outflow_{st_percSmax[iter_vol]}MCM'] = outflow_da
+        else:
+            forecast_outflow[f'ST_sarea_{st_percSmax[iter_vol]}%'] = sarea_da
+            forecast_outflow[f'ST_delH_{st_percSmax[iter_vol]}%'] = delH_da
+            forecast_outflow[f'ST_delS_{st_percSmax[iter_vol]}%'] = delS_da
+            forecast_outflow[f'ST_elevation_{st_percSmax[iter_vol]}%'] = elevation_da
+            forecast_outflow[f'ST_outflow_{st_percSmax[iter_vol]}%'] = outflow_da    
+    return forecast_outflow
+
+def forecast_outflow_for_res(
+        base_date, 
+        forecast_lead_time,
+        forecast_inflow_fp,
+        evap_fp,
+        sarea_fp,
+        aec_fp,
+        dels_scenario = None, 
+        cust_wl = None, 
+        s_max = None, 
+        rule_curve = None, 
+        st_percSmax = [0.5, 1, 2.5], 
+        cust_st = None, 
+        output_path = None
+    ):
+    ''' 
+    Generates forecasted outflow for RAT given forecasted inflow time series data based on specific delS scenarios.
+    Returns an xarray dataset containing forecasted results.
+    
+    Parameters
+    ----------
+    forecast_inflow_fp: str
+        Path to forecasted inflow netcdf file
+    base_date: str or datetime object
+        Start date of forecasted outflow estimation. This is typically the end date of hindcast RAT run + 1.
+    rat_folder: str
+        Path to current working RAT folder. For obtaining ancillary data such as evaporation, surface area, Area Elevation Curve.
+    dels_scenario: str
+        Specifies the delS scenario to use. If None, returns results for all scenarios
+        values: 
+            GC - Gates Closed scenario: All inflow will be accumulated as change in storage. Final surface area, storage state, water level is returned
+            GO - Gates Open scenario: All inflow will be released as outflow. Outflows returned 
+            CL - Custom Water Level scenario: Inflow will be accumulated till desired water level, then released as outflow
+            RC - Rule Curve scenario: Outflow will be calculated depending on the permissible rule curve based change in storage
+            ST - storage change scenario: Outflows will be calculated based on permissible storage values provided 
+    cust_wl: float
+        Specifies the value for custom water level in meters above sea level. Only used if dels_scenario = CL or None.
+    s_max: float
+        Maximum storage capacity of the reservoir in Million cubic meters (MCM)
+    rule_curve: str
+        Path to rule curve file in .csv with headers (Month, S/Smax)   
+    st_percSmax: list
+        A list of values of storages as percentage of Smax
+    cust_st: float
+        User defined custom storage value in Million cubic meters (MCM). Used if dels_scenario = ST or None  
+    output_path: str
+        If provided, generates a netcdf file at the specified path containing the forecasted outflow results.
```

Comment:
There is a mismatch between docstring and the parameters that the function takes. For example, I think rat_folder has later been changed to eval_fp, area_fp and so on.

Reply:
Thanks for catching that, I have updated the docstring.